quiz/questions.rst

```diff
@@ -1,62 +1,85 @@
+4) What is the difference between an Array and a Hash (aka Dictionary)?
 
 
-5) What is the difference between a Class variable and an Instance variable?
+5) Tabs or Spaces? Why?
 
 
-6) What is the difference between Object Oriented and Procedural programming? Give an example of where Procedural would work better than Object Oriented.
+6) Given the following database table, write an SQL query that will return the
+animal name and food of the three animals with the highest populations.
 
+	|id     |animal    |food        |population |
+	|-------|----------|------------|-----------|
+	|35     |ocelot    |mouse       |1200000    |
+	|36     |dingo     |rabbit      |16000000   |
+	|41     |capybara  |grass       |400000     |
+	|52     |remora    |blood       |82000000   |
+	|54     |emu       |fruit       |725000     |
+	|63     |gecko     |insects     |9000000    |
+	|68     |earwig    |lettuce     |420000000  |
 
-7) In a relational database, what is a "join table", and why might you need one?
 
+7) What is your favorite editor, and why?
 
-8) Given the following database table, write an SQL query that will return a list of the colors of the three tallest plants.
 
-	|id    |plant    |color       |height |
-	|------|---------|------------|-------|
-	|1     |tree     |red         |12     |
-	|2     |bush     |blue        |2      |
-	|3     |vine     |green       |30     |
-	|4     |flower   |yellow      |500    |
-	|5     |hedge    |purple      |8      |
-	|6     |moss     |white       |1      |
+8) What is Recursion, and how would you use it?
 
 
-9) What is Recursion, and how would you use it?
+9) Code Debugging:
 
+	In the scripts directory in this repository is a file called names.rb. This
+  code is supposed to print a numbered list of names, email addresses, and a
+  weighted score - but there are a number of problems with the logic.
 
-10) Code Debugging: 
+	Debug this script, fix the code as needed and add comments in the code
+  describing what was wrong and how you fixed it. The code is written in Ruby,
+  but the errors are not due to ruby syntax, and you should not need to know the
+  ruby language to spot the errors. You can run it anywhere you have access to
+  ruby by typing "ruby names.rb". Feel free to use Google or other resources to
+  help you understand the syntax if you are not familiar.
 
-	In the scripts directory in this repository is a file called debug_this.py. This code is supposed to print a numbered list of names, email addresses, and a computed 'Z Number' - but there are a number of problems with the logic. 
 
-	Debug this script, fix the code as needed and add comments in the code describing what was wrong and how you fixed it. The code is written in Python 2.7, but the errors are not due to python syntax, and you should not need to know the python language to spot the errors. You can run it anywhere you have access to python by typing "python debug_this.py". 
-
-
-11) Code sample:
+10) Code sample:
```

Comment:
It would be good to add a note saying that a machine can have multiple dependencies, in which case it will be listed multiple times with the same name.